engines/tika/src/main/java/org/alfresco/transform/tika/transformers/Tika.java

```diff
@@ -0,0 +1,422 @@
+    {
+        try
+        {
+            ContentHandler handler;
+            if (MIMETYPE_TEXT_PLAIN.equals(targetMimetype))
+            {
+                handler = new BodyContentHandler(output);
+            }
+            else
+            {
+                SAXTransformerFactory factory = (SAXTransformerFactory) SAXTransformerFactory.newInstance();
+                TransformerHandler transformerHandler;
+                transformerHandler = factory.newTransformerHandler();
+                transformerHandler.getTransformer().setOutputProperty(OutputKeys.INDENT, "yes");
+                transformerHandler.setResult(new StreamResult(output));
+                handler = transformerHandler;
+
+                if (MIMETYPE_HTML.equals(targetMimetype))
+                {
+                    transformerHandler.getTransformer().setOutputProperty(OutputKeys.METHOD, HTML);
+                    return new ExpandedTitleContentHandler(transformerHandler);
+                }
+                else if (MIMETYPE_XHTML.equals(targetMimetype) ||
+                         MIMETYPE_XML.equals(targetMimetype))
+                {
+                    transformerHandler.getTransformer().setOutputProperty(OutputKeys.METHOD, XML);
+                }
+                else if (MIMETYPE_TEXT_CSV.equals(targetMimetype))
+                {
+                    handler = new CsvContentHandler(output);
+                }
+                else
+                {
+                    throw new IllegalArgumentException("Invalid target mimetype " + targetMimetype);
+                }
+            }
+            return handler;
+        }
+        catch (TransformerConfigurationException e)
+        {
+            throw new IllegalStateException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * A wrapper around the normal Tika BodyContentHandler for CSV rather encoding than tab separated.
+     */
+    protected static class CsvContentHandler extends BodyContentHandler
+    {
+        private static final char[] comma = new char[]{','};
+        private static final Pattern all_nums = Pattern.compile("[\\d\\.\\-\\+]+");
+
+        private boolean inCell = false;
+        private boolean needsComma = false;
+
+        protected CsvContentHandler(Writer output)
+        {
+            super(output);
+        }
+
+        @Override
+        public void ignorableWhitespace(char[] ch, int start, int length)
+            throws SAXException
+        {
+            if (length == 1 && ch[0] == '\t')
+            {
+                // Ignore tabs, as they mess up the CSV output
+            }
+            else
+            {
+                super.ignorableWhitespace(ch, start, length);
+            }
+        }
+
+        @Override
+        public void characters(char[] ch, int start, int length)
+            throws SAXException
+        {
+            if (inCell)
+            {
+                StringBuffer t = new StringBuffer(new String(ch, start, length));
```

Comment:
I can't see any reason to use `StringBuffer` over `StringBuider`. Unless the synchroization of `StringBuffer` is required, `StringBuilder` would be better as it will have a smaller impact on performance being unsynchronized.

[Synchronized classes Vector, Hashtable, Stack and StringBuffer should not be used (java:S1149)](https://rules.sonarsource.com/java/type/Code%20Smell/RSPEC-1149)

Reply:
I'm not touching this original code. Not worth the risk. You are correct to not use StingBuffer in new code.